### DISCUSSION TITLE
VS2022 Pet Peeve - Default namespace for unknown type is super unhelpful!

### DISCUSSION THREAD

**Discussion by trinith** [MonoGame/MonoGame#8428] [General]
This isn't MonoGame's fault, I'm just wondering if anybody else has bumped up against this and what you use, if anything, to make it less frustrating for you...

When I'm using a type I haven't used before, VS identifies that it doesn't know what it is and is helpful enough to give you a fix. I can just press ctrl+. and then enter and it will give me what it thinks is the best namespace to resolve the type. Prior to VS2022 this always gave me the MonoGame one but apparently in VS2022, it's changed. In the case of `Vector2` it just really wants me to use the `System.Numerics` version, which is infuriating because this keyboard combo is muscle memory by this point. 

![image](https://github.com/user-attachments/assets/49d247d9-9aed-45cf-a6a5-437b9e19fcca)

I'm not terribly deep in this project and I've alrady trolled myself many times by grabbing the wrong using because the default is not the one I want. You'd think VS would be smart enough to be like, "Hey, this person has used X namespace to resolve Y type about 100 times already, maybe I'll just go with that!" but noooooo!

Ok, rant over... but does anybody else enounter this and if so, how do you work around it?

**Comment by Brett208** [comment]
@trinith I have the same problem. 

A good way to sidestep the problem is to use a global using file. 

I'll create a GlobalUsings.cs file and put in the following content:
```CSharp
global using Microsoft.Xna.Framework;
global using Microsoft.Xna.Framework.Content;
global using Microsoft.Xna.Framework.Graphics;
global using System;
global using System.Collections.Generic;
```

That tends to grab the bulk of the using statements up front and sort of hides the problem.

~Brett

**Reply by trinith** [reply]
I'll look into this, I didn't know global using was a thing :)

**Comment by stromkos** [comment]
The first thing that I do when creating a new Class is to copy the `using` statements from `Game1.cs` after adding `using System;` to allow access to `Math.` without qualification.

The use of `global` becomes a problem when you need to use native structs like "Point" or "Color" in `Program.cs` or `Activity.cs`.

I went as far as creating a new file template `MGClass` that included the common code for a new class to minimize boilerplate code.
